RecordingApp/app/src/main/java/wycliffeassociates/recordingapp/ProjectManager/activities/ActivityProjectManager.java

```diff
@@ -411,29 +426,44 @@ public void setCurrentFile(String currentFile) {
     }
 
     @Override
-    public void onPause(){
+    public void onPause() {
         super.onPause();
         dismissExportProgressDialog();
-        dismissDatabaseResyncProgressDialog();
     }
 
-    private void dismissExportProgressDialog(){
-        if(mPd != null && mPd.isShowing()){
+    private void dismissExportProgressDialog() {
+        if (mPd != null && mPd.isShowing()) {
             mPd.dismiss();
             mPd = null;
         }
     }
 
-    private void dismissDatabaseResyncProgressDialog(){
-        if(mDatabaseProgressDialog != null && mDatabaseProgressDialog.isShowing()){
-            mDatabaseProgressDialog.dismiss();
-            mDatabaseProgressDialog = null;
-        }
-    }
-
     @Override
     public void delegateExport(Export exp) {
         exp.setFragmentContext(mExportTaskFragment);
         mExportTaskFragment.delegateExport(exp);
     }
+
+    @Override
+    public void delegateSourceAudio(Project project) {
+        mProjectToExport = project;
+        Intent intent = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType("*/*");
+        mSourceAudioFile = new File(getFilesDir(), project.getTargetLanguage() + "_" + project.getSource() + "_" + project.getSlug() + ".tr");
+        intent.putExtra(Intent.EXTRA_TITLE, mSourceAudioFile.getName());
+        startActivityForResult(intent, SAVE_SOURCE_AUDIO_REQUEST);
+    }
+
+    @Override
+    public void onTaskComplete(int taskTag, int resultCode) {
+        if (resultCode == TaskFragment.STATUS_OK) {
+            if (taskTag == DATABASE_RESYNC_TASK) {
+                ProjectDatabaseHelper db = new ProjectDatabaseHelper(this);
+                mNumProjects = db.getNumProjects();
+                mDbResyncing = false;
+                initializeViews();
+            }
+        }
```

Comment:
onError() and onCancel() is included in the interface because it makes sense that a task could be cancelled by the user and therefore need to communicate that result with the TaskFragment and the Task itself. Handling of this is a responsibility of the subclassed task.

In this particular case, the current tasks implemented have no onError or onCancel capabilities (Errors will throw exceptions and cause crashes), meaning nothing currently has the ability to throw a STATUS_ERROR or STATUS_CANCEL. An else clause should therefore be a responsibility of adding a task to the task fragment that can be cancelled or error out. 

onError may need to be modified anyway if something uses it in the future. It's all mostly just stuff to preempt future use cases.